src/promptflow/promptflow/_core/tool_meta_generator.py

```diff
@@ -137,6 +137,7 @@ def _parse_tool_from_function(f, initialize_inputs=None, gen_custom_type_conn=Fa
             initialize_inputs=initialize_inputs,
             gen_custom_type_conn=gen_custom_type_conn,
             skip_prompt_template=skip_prompt_template,
+            add_ui_hints=tool_type == ToolType.CUSTOM_LLM,
```

Comment:
Why we did this?

Reply:
Changed to use a function `_should_add_index` to control it, thanks.